File: apps/api/app/routers/health.py
"""Health check endpoints."""
from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session
from sqlalchemy import text
from ..core.database import get_db

router = APIRouter(prefix="/health", tags=["health"])


@router.get("")
async def health_check():
    """Basic health check endpoint."""
    return {"status": "healthy", "service": "koosdoos-api"}


@router.get("/db")
def database_health(db: Session = Depends(get_db)):
    """Database connectivity health check."""
    try:
        db.execute(text("SELECT 1"))
        return {"status": "healthy", "database": "connected"}
    except Exception as e:
        return {"status": "unhealthy", "database": "disconnected", "error": str(e)}

File: apps/web/src/lib/api/health.ts
/**
 * Health Check API Service
 */
import api from "../api-client";
import type { HealthResponse, HealthDbResponse } from "./types";

/**
 * Basic health check
 */
export async function getHealth(): Promise<HealthResponse> {
  return api.get<HealthResponse>("/health");
}

/**
 * Health check with database connectivity
 */
export async function getHealthDb(): Promise<HealthDbResponse> {
  return api.get<HealthDbResponse>("/health/db");
}

export const healthApi = {
  getHealth,
  getHealthDb,
};
